File: 2014/20141130-gocon_2014_autumn/double_test.rb
require './double'
require 'stringio'
require 'test/unit'

class DoubleTest < Test::Unit::TestCase
  def test_double
    stdin  = StringIO.new("hoge\n", "r")
    stdout = StringIO.new("", "w")

    double(stdin, stdout)

    assert_equal("hoge\nhoge\n", stdout.string)
  end
end

File: 2014/11/30/gocon_2014_autumn/double_test.go
package main

import (
	"bytes"
	"os/exec"
	"testing"
)

func TestDouble(t *testing.T) {
	stdin := bytes.NewBufferString("hoge\n")
	stdout := new(bytes.Buffer)

	Double(stdin, stdout)

	if stdout.String() != "hoge\nhoge\n" {
		t.Fatal("Not matched")
	}
}

func TestCommand(t *testing.T) {
	cmd := exec.Command("go", "run", "double.go")
	cmd.Stdin = bytes.NewBufferString("hoge\n")
	stdout := new(bytes.Buffer)
	cmd.Stdout = stdout

	_ = cmd.Run()

	if stdout.String() != "hoge\nhoge\n" {
		t.Fatal("Not matched")
	}
}
